util/reggen/gen_cheader.py

```diff
@@ -188,6 +205,43 @@ def gen_cdefines_module_params(outstr, module_data, module_name,
     genout(outstr, '\n')
 
 
+def gen_multireg_field_defines(outstr, regname, field, subreg_num, regwidth,
+                               existing_defines):
+    field_width = field['bitinfo'][1]
+    fields_per_reg = int(regwidth / field_width)
+
+    define_name = regname + '_' + as_define(field['name'] + "_FIELD_WIDTH")
+    define = gen_define(define_name, [], str(field_width), existing_defines)
+    genout(outstr, define)
+
+    define_name = regname + '_' + as_define(field['name'] + "_FIELDS_PER_REG")
+    define = gen_define(define_name, [], str(fields_per_reg), existing_defines)
+    genout(outstr, define)
+
+    define_name = regname + '_' + as_define(field['name'] + "_REG_NUM")
```

Comment:
Do you think it might be clearer if the suffix here is `_MULTIREG_COUNT`?

Reply:
Sure, I will update it.

Reply:
I have also removed the field name from the count define, as it doesn't make any sense to have it there. So `RV_PLIC_IP_P_MULTIREG_COUNT` becomes `RV_PLIC_IP_MULTIREG_COUNT`.